### PR TITLE
doc: Clarify configure flags for optional modules

### PR DESCRIPTION
**Pull request by robot-dreams**
Minor doc cleanup as a follow-up to https://github.com/bitcoin-core/secp256k1/pull/995#discussion_r835105980

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -141,19 +141,19 @@ AC_ARG_ENABLE(examples,
     [SECP_SET_DEFAULT([enable_examples], [no], [yes])])
 
 AC_ARG_ENABLE(module_ecdh,
-    AS_HELP_STRING([--enable-module-ecdh],[enable ECDH shared secret computation]), [],
+    AS_HELP_STRING([--enable-module-ecdh],[enable ECDH shared secret computation [default=no]]), [],
```
Comment (real-or-random) [inline configure.ac:144]: maybe this is the time to reword this to "enable ECDH module"

File: README.md [hunk 2]
```diff
@@ -68,12 +68,15 @@ libsecp256k1 is built using autotools:
     $ make check  # run the test suite
     $ sudo make install  # optional
 
+To compile optional modules (such as Schnorr), you need to run `./configure` with additional flags (such as `--enable-module-schnorr`). Run `./configure --help` to see the full list of available flags.
```
Comment (real-or-random) [inline README.md:71]:
```suggestion
To compile optional modules (such as Schnorr signatures), you need to run `./configure` with additional flags (such as `--enable-module-schnorrsig`). Run `./configure --help` to see the full list of available flags.
```

Also below.

"Schnorr" is a person, so I think "enable Schnorr" is very strange. I keep insisting on this point. (But I don't always win  -- https://github.com/rust-bitcoin/rust-secp256k1/pull/327)

File: README.md [hunk 3]
```diff
@@ -68,12 +68,15 @@ libsecp256k1 is built using autotools:
...
 
+To compile optional modules (such as Schnorr signatures), you need to run `./configure` with additional flags (such as `--enable-module-schnorrsig`). Run `./configure --help` to see the full list of available flags.
+
 Usage examples
 -----------
   Usage examples can be found in the [examples](examples) directory. To compile them you need to configure with `--enable-examples`.
   * [ECDSA example](examples/ecdsa.c)
-  * [Schnorr Signatures example](examples/schnorr.c)
+  * [Schnorr signatures example](examples/schnorr.c)
   * [Deriving a shared secret(ECDH) example](examples/ecdh.c)
```
Comment (real-or-random) [inline README.md:78]:
```suggestion
  * [Deriving a shared secret (ECDH) example](examples/ecdh.c)
```

Not that we're on it. (Sorry for not pointing this out earlier. :P)

**Review by jonasnick — COMMENTED** [review]
ACK 55512d30b7921bc46247a78be1da98108f243c1c
